pkg/services/secrets/kvstore/kvstore.go

```diff
@@ -14,29 +15,31 @@ const (
 )
 
 func ProvideService(sqlStore sqlstore.Store, secretsService secrets.Service, remoteCheck UseRemoteSecretsPluginCheck) SecretsKVStore {
+	var store SecretsKVStore
 	logger := log.New("secrets.kvstore")
+	store = &secretsKVStoreSQL{
+		sqlStore:       sqlStore,
+		secretsService: secretsService,
+		log:            logger,
+		decryptionCache: decryptionCache{
+			cache: make(map[int64]cachedDecrypted),
+		},
+	}
 	if remoteCheck.ShouldUseRemoteSecretsPlugin() {
 		logger.Debug("secrets kvstore is using a remote plugin for secrets management")
 		secretsPlugin, err := remoteCheck.GetPlugin()
 		if err != nil {
 			logger.Error("plugin client was nil, falling back to SQL implementation")
 		} else {
-			return &secretsKVStorePlugin{
+			store = &secretsKVStorePlugin{
 				secretsPlugin:  secretsPlugin,
 				secretsService: secretsService,
 				log:            logger,
 			}
 		}
 	}
 	logger.Debug("secrets kvstore is using the default (SQL) implementation for secrets management")
-	return &secretsKVStoreSQL{
-		sqlStore:       sqlStore,
-		secretsService: secretsService,
-		log:            logger,
-		decryptionCache: decryptionCache{
-			cache: make(map[int64]cachedDecrypted),
-		},
-	}
+	return NewCachedKVStore(store, 5*time.Second, 10*time.Second)
```

Comment:
Nit. Cleanup interval might be a bit low, but hard to tell without testing it. Does the cache library have any recommendations?

Reply:
I increased the clean up interval to 5 minutes.